src/wallet/direct_send.rs

```diff
@@ -9,12 +9,25 @@ use crate::wallet::{wallet::UTXOSpendInfo, SwapCoin};
 
 use super::{error::WalletError, fidelity::get_locktime_from_index, Wallet};
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Eq)]
 pub enum SendAmount {
     Max,
     Amount(Amount),
 }
 
+impl PartialEq for SendAmount {
```

Comment:
I don't see why you can't just derive this? Put it above the `#derive(..)` list on top of struct definition, it will work. No need for this much code. 

You will have to derive these usually in some very edge cases of struct definitions..

Reply:
My bad... I thought it won't work... Let me correct it... Thanks for review...